tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -65,6 +73,40 @@ protected void addBadWords() {
 	}
 
 	@Override
+	protected void genFile(Grammar g, ST outputFileST, String fileName) {
+		super.genFile(g, outputFileST, fileName);
+		if (DO_GOFMT && !fileName.startsWith(".") /* criterion taken from gofmt*/ && fileName.endsWith(".go")) {
+			gofmt(new File(getCodeGenerator().tool.getOutputDirectory(g.fileName), fileName));
+		}
+	}
+
+	private void gofmt(File fileName) {
+		// Optimistically run gofmt. If this fails, it doesn't matter at this point. Wait for termination though,
+		// because "gofmt -w" uses ioutil.WriteFile internally, which means it literally writes in-place with O_TRUNC.
+		// That could result in a race. (Why oh why doesn't it do tmpfile + rename?)
+		try {
+			ProcessBuilder gofmtBuilder = new ProcessBuilder("gofmt", "-w", "-s", fileName.getPath());
+			gofmtBuilder.redirectErrorStream(true);
+			Process gofmt = gofmtBuilder.start();
+			InputStream stdout = gofmt.getInputStream();
+			byte[] buf = new byte[1 << 10];
+			for (int l = 0; l > -1; l = stdout.read(buf)) {
+				// There should not be any output that exceeds the implicit output buffer. In normal ops there should be
+				// zero output. In case there is output, blocking and therefore killing the process is acceptable. This
+				// drains the buffer anyway to play it safe.
+
+				// dirty debug (change -w above to -d):
+				// System.err.write(buf, 0, l);
```

Comment:
Won't this only print out the result from the last read call? Would a BufferedReader work here? If this isn't meant to ship back to antlr/antlr4, we should tag it with a TODO to remove it.